Feature/datastructures, fixes #80

lukasboehm97 opened the pull request:
### Problems

- Alternative words should be editable within a document
- documents and training sets have an n:m relationships
- documents should be added dynamically within a training set

### Solution

- Included alternative word module as "nested module" in the document module
- implemented a many to many field, pointing from the training set to the respective documents. Hence, the foreign key field of the document module was removed
- adapt url patterns

### Further tasks

- the dynamic selection of documents in the training sets is not user friendly. Thus, a forms.py should be created in order to modify the UI (future issue)

svenseeberg commented:
@lukasboehm97 you deleted existing migrations. Is this intentional? This will prevent migrating existing databases to the new state. In the end this means the database content will be lost.

svenseeberg commented on src/vocgui/models.py:
```diff
@@ -28,7 +28,9 @@ class Document(models.Model):  # pylint: disable=R0903
     """
     word = models.CharField(max_length=255)
-    article = models.CharField(max_length=255, choices=[('der', 'der'), ('das', 'das'), ('die', 'die'), ('die (Plural)', 'die (Plural)')], default='')
+    article = models.CharField(max_length=255, choices=[('der', 'der'), ('das', 'das'), 
```
The `choices` list should be moved into a module static and then be reused here and for the alternative word as well.

lukasboehm97 replied:
alright

svenseeberg reviewed (changes requested):
Please do not remove existing migrations. Only do so, if there is really a benefit of doing so. This needs a very good reason. If there is one, please state it in the Pull Request.

lukasboehm97 commented:
I'll then will use the existing migrations and modify them manually. I changed the relationship of training sets and documents (from 1:n to n:m). This behaviour is not support by Django (since an existing migration file first implements a foreign key, which than is overwritten by another migration file at a later stage. This throws an error).

lukasboehm97 commented:
@svenseeberg I now implemented the feature without deleting all existing migrations. Furthermore, I added an static module to handle global settings.
